extra_data/reader.py

```diff
@@ -1197,37 +1197,41 @@ def info(self, details_for_sources=()):
             span_txt = '{}.{}'.format(datetime.timedelta(seconds=seconds),
                                       int(deciseconds))
 
-        detector_modules = {}
-        for source in self.detector_sources:
-            name, modno = DETECTOR_SOURCE_RE.match(source).groups((1, 2))
-            detector_modules[(name, modno)] = source
-
-        # A run should only have one detector, but if that changes, don't hide it
-        detector_name = ','.join(sorted(set(k[0] for k in detector_modules)))
-
         # disp
         print('# of trains:   ', train_count)
         print('Duration:      ', span_txt)
         print('First train ID:', first_train)
         print('Last train ID: ', last_train)
         print()
 
-        print("{} detector modules ({})".format(
-            len(self.detector_sources), detector_name
-        ))
-        if len(detector_modules) > 0:
-            # Show detail on the first module (the others should be similar)
-            mod_key = sorted(detector_modules)[0]
-            mod_source = detector_modules[mod_key]
-            dinfo = self.detector_info(mod_source)
-            module = ' '.join(mod_key)
-            dims = ' x '.join(str(d) for d in dinfo['dims'])
-            print("  e.g. module {} : {} pixels".format(module, dims))
-            print("  {}".format(mod_source))
-            print("  {} frames per train, up to {} frames total".format(
-                dinfo['frames_per_train'], dinfo['total_frames']
+        if not details_for_sources:
+            # Include summary section for multi-module detectors unless
+            # source details are enabled.
+
+            detector_modules = {}
+            for source in self.detector_sources:
+                name, modno = DETECTOR_SOURCE_RE.match(source).groups((1, 2))
+                detector_modules[(name, modno)] = source
+
+            # A run should only have one detector, but if that changes, don't hide it
+            detector_name = ','.join(sorted(set(k[0] for k in detector_modules)))
+
+            print("{} detector modules ({})".format(
```

Comment:
```suggestion
            print("{} XTDF detector modules ({})".format(
```

While we're looking at this code, let's take a moment to make the messages a bit more accurate.